llm_bioinformatics_research/src/UserProfile/UserProfile.js

```diff
@@ -62,7 +62,17 @@ const UserProfile = () => {
     if (file) {
       const reader = new FileReader();
       reader.onloadend = () => {
-        setUser({ ...user, image: reader.result });
+        const img = new Image();
+        img.src = reader.result;
+        img.onload = () => {
+          const canvas = document.createElement('canvas');
+          const ctx = canvas.getContext('2d');
+          canvas.width = 100;
+          canvas.height = 100;
+          ctx.drawImage(img, 0, 0, 100, 100);
```

Comment:
When I uploaded images that were a few MBs in size, there were no issues so this is working correctly.